src/main/java/com/epam/ta/reportportal/core/widget/content/loader/ComponentHealthCheckContentLoader.java

```diff
@@ -0,0 +1,167 @@
+				.verify(ErrorType.UNABLE_LOAD_WIDGET_CONTENT, "Incorrect level definition");
+
+		Filter launchesFilter = GROUP_FILTERS.apply(filterSortMapping.keySet());
+		Sort launchesSort = GROUP_SORTS.apply(filterSortMapping.values());
+
+		boolean latestMode = WidgetOptionUtil.getBooleanByKey(LATEST_OPTION, widgetOptions);
+
+		Filter testItemFilter = Filter.builder()
+				.withTarget(TestItem.class)
+				.withCondition(getTestItemCondition(contentFields, attributeValues))
+				.build();
+
+		String currentLevelKey = ofNullable(contentFields.get(attributeValues.size())).orElseThrow(() -> new ReportPortalException(ErrorType.UNABLE_LOAD_WIDGET_CONTENT,
+				"Current level key should be not null"
+		));
+
+		List<ComponentHealthCheckContent> content = widgetContentRepository.componentHealthCheck(launchesFilter,
+				launchesSort,
+				latestMode,
+				limit,
+				testItemFilter,
+				currentLevelKey
+		);
+
+		return CollectionUtils.isNotEmpty(content) ? Collections.singletonMap(RESULT, content) : emptyMap();
+	}
+
+	private void validateContentFields(List<String> contentFields) {
+		BusinessRule.expect(contentFields, CollectionUtils::isNotEmpty)
+				.verify(ErrorType.UNABLE_LOAD_WIDGET_CONTENT, "No keys were specified");
+		BusinessRule.expect(contentFields, cf -> cf.size() <= MAX_LEVEL_NUMBER)
+				.verify(ErrorType.UNABLE_LOAD_WIDGET_CONTENT, "Keys number is incorrect. Maximum keys count = " + MAX_LEVEL_NUMBER);
+	}
+
+	private void validateAttributeValues(List<String> attributeValues) {
+		attributeValues.forEach(value -> BusinessRule.expect(value, Objects::nonNull)
+				.verify(ErrorType.BAD_REQUEST_ERROR, "Attribute value should be not null"));
+	}
+
+	private ConvertibleCondition getTestItemCondition(List<String> attributeKeys, List<String> attributeValues) {
+
+		List<ConvertibleCondition> conditions = Lists.newArrayList(FilterCondition.builder()
+						.eq(CRITERIA_HAS_STATS, String.valueOf(Boolean.TRUE))
+						.build(),
+				FilterCondition.builder().eq(CRITERIA_HAS_CHILDREN, String.valueOf(Boolean.FALSE)).build(),
+				FilterCondition.builder()
+						.withCondition(Condition.IN)
+						.withNegative(false)
+						.withSearchCriteria(CRITERIA_TYPE)
+						.withValue(String.join(",",
+								TestItemTypeEnum.STEP.name(),
+								TestItemTypeEnum.SCENARIO.name(),
+								TestItemTypeEnum.TEST.name()
+						))
+						.build(),
+				FilterCondition.builder()
+						.withCondition(Condition.EXISTS)
+						.withNegative(true)
+						.withSearchCriteria(CRITERIA_RETRY_PARENT_ID)
+						.withValue(String.valueOf(0L))
+						.build(),
+				FilterCondition.builder()
+						.withCondition(Condition.NOT_EQUALS)
+						.withNegative(false)
+						.withSearchCriteria(CRITERIA_STATUS)
+						.withValue(StatusEnum.IN_PROGRESS.name())
+						.build()
+		);
+
+		if (CollectionUtils.isNotEmpty(attributeValues)) {
+			String attributeCriteria = IntStream.range(0, attributeValues.size()).boxed().map(index -> {
```

Comment:
can be used mapToObj(...) instead of boxed

Reply:
replaced